### ISSUE TITLE
Kernel Panic

### ISSUE THREAD

**Issue by skinless8** [donmelton/video_transcoding#101]
I'm consistently getting kernel panic while doing transcode_video

From a glance, it looks like the culprit is `HandbrakeCLI`. I have tried the version that come with `brew cask install handbrakecli` and also from nightly build with the same problem.

Any pointers on what to do next to resolve this issue?

Here's .panic file content:
```
** Panic Report ***
panic(cpu 4 caller 0xffffff801c5e629d): "TLB invalidation IPI timeout: " "CPU(s) failed to respond to interrupts, unresponsive CPU bitmap: 0x4, NMIPI acks: orig: 0x0, now: 0x0"@/Library/Caches/com.apple.xbs/Sources/xnu/xnu-3789.21.3/osfmk/x86_64/pmap.c:2724
Backtrace (CPU 4), Frame : Return Address
0xffffff920bee3ae0 : 0xffffff801c4f368c
0xffffff920bee3b60 : 0xffffff801c5e629d
0xffffff920bee3c00 : 0xffffff801c5ed01e
0xffffff920bee3d00 : 0xffffff801c5edfca
0xffffff920bee3d70 : 0xffffff801c57b0b8
0xffffff920bee3eb0 : 0xffffff801c57a9bc
0xffffff920bee3ee0 : 0xffffff801c4e8296
0xffffff920bee3f10 : 0xffffff801c5f26ae
0xffffff920bee3fb0 : 0xffffff801c4a5f66

BSD process name corresponding to current thread: HandBrakeCLI
Boot args: dart=0

Mac OS version:
16B2555

Kernel version:
Darwin Kernel Version 16.1.0: Thu Oct 13 21:26:57 PDT 2016; root:xnu-3789.21.3~60/RELEASE_X86_64
Kernel UUID: 8941AC1C-B084-37DE-8A34-4CE638C5CFC9
Kernel slide:     0x000000001c200000
Kernel text base: 0xffffff801c400000
__HIB  text base: 0xffffff801c300000
System model name: iMac14,2 (Mac-27ADBB7B4CEE8E61)

System uptime in nanoseconds: 110326975528254
last loaded kext at 69875268567274: com.apple.driver.AppleXsanScheme    3 (addr 0xffffff7f9f71b000, size 32768)
last unloaded kext at 69940288954398: com.apple.driver.AppleXsanScheme  3 (addr 0xffffff7f9f71b000, size 32768)
loaded kexts:
com.aladdin.kext.aksfridge      1.0.2
org.hwsensors.driver.GPUSensors 1723
com.intel.driver.EnergyDriver   2.0
org.hwsensors.driver.CPUSensors 1723
org.hwsensors.driver.LPCSensors 1723
com.makemkv.kext.daspi  1.5
org.hwsensors.driver.ACPISensors        1723
com.TrustedData.driver.VendorSpecificType00     1.8.0
com.insanelymac.RealtekRTL8111  2.2.1
org.rehabman.driver.FakePCIID.XHCIMux   1.3.2
org.rehabman.driver.FakePCIID   1.3.2
org.netkas.driver.FakeSMC       1723
com.rehabman.driver.USBInjectAll        0.5.14
com.apple.driver.AudioAUUC      1.70
com.apple.filesystems.autofs    3.0
com.apple.driver.AppleBluetoothMultitouch       92
com.apple.driver.AppleHDA       278.23
com.apple.driver.AppleUpstreamUserClient        3.6.4
com.apple.driver.pmtelemetry    1
com.apple.driver.AppleMCCSControl       1.2.13
com.apple.iokit.IOUserEthernet  1.0.1
com.apple.iokit.IOBluetoothSerialManager        5.0.1f7
com.apple.GeForce       10.1.2
com.apple.Dont_Steal_Mac_OS_X   7.0.0
com.apple.driver.AppleHV        1
com.apple.driver.AppleOSXWatchdog       1
com.apple.iokit.BroadcomBluetoothHostControllerUSBTransport     5.0.1f7
com.apple.driver.AppleIntelSlowAdaptiveClocking 4.0.0
com.apple.driver.ACPI_SMC_PlatformPlugin        1.0.0
com.apple.AppleFSCompression.AppleFSCompressionTypeDataless     1.0.0d1
com.apple.AppleFSCompression.AppleFSCompressionTypeZlib 1.0.0
com.apple.BootCache     40
com.apple.filesystems.hfs.kext  366.1.1
com.apple.iokit.SCSITaskUserClient      394
com.apple.iokit.IOAHCIBlockStorage      295.20.1
com.apple.driver.AirPort.Brcm4360       1110.4.1a8
com.apple.driver.AppleAHCIPort  326
com.apple.driver.AppleRTC       2.0
com.apple.driver.AppleACPIButtons       5.0
com.apple.driver.AppleHPET      1.8
com.apple.driver.AppleSMBIOS    2.1
com.apple.driver.AppleAPIC      1.7
com.apple.nke.applicationfirewall       172
com.apple.security.quarantine   3
com.apple.security.TMSafetyNet  8
com.apple.driver.AppleBluetoothHIDKeyboard      197
com.apple.kext.triggers 1.0
com.apple.driver.IOBluetoothHIDDriver   5.0.1f7
com.apple.driver.AppleMultitouchDriver  368.4
com.apple.driver.AppleInputDeviceSupport        76.4
com.apple.driver.DspFuncLib     278.23
com.apple.kext.OSvKernDSPLib    525
com.apple.driver.AppleSMBusController   1.0.14d1
com.apple.iokit.IOSMBusFamily   1.1
com.apple.iokit.IOSurface       153.1
com.apple.iokit.IOSerialFamily  11
com.apple.iokit.IOAcceleratorFamily2    288.15
com.apple.nvidia.driver.NVDAGK100Hal    10.1.2
com.apple.nvidia.driver.NVDAResman      10.1.2
com.apple.AppleGraphicsDeviceControl    3.13.74
com.apple.iokit.IONDRVSupport   2.4.1
com.apple.iokit.IOBluetoothHostControllerUSBTransport   5.0.1f7
com.apple.iokit.IOBluetoothHostControllerTransport      5.0.1f7
com.apple.iokit.IOBluetoothFamily       5.0.1f7
com.apple.driver.AppleHDAController     278.23
com.apple.iokit.IOGraphicsFamily        2.4.1
com.apple.iokit.IOHDAFamily     278.23
com.apple.iokit.IOAudioFamily   205.11
com.apple.vecLib.kext   1.2.0
com.apple.iokit.IOSlowAdaptiveClockingFamily    1.0.0
com.apple.driver.AppleSMC       3.1.9
com.apple.driver.IOPlatformPluginLegacy 1.0.0
com.apple.driver.IOPlatformPluginFamily 6.0.0d8
com.apple.driver.AppleHIDKeyboard       197
com.apple.iokit.IOSCSIBlockCommandsDevice       394
com.apple.iokit.IOUSBMassStorageDriver  131.1.1
com.apple.driver.usb.IOUSBHostHIDDevice 1.1
com.apple.driver.usb.networking 5.0.0
com.apple.driver.usb.AppleUSBHostCompositeDevice        1.1
com.apple.driver.usb.AppleUSBHub        1.1
com.apple.filesystems.hfs.encodings.kext        1
com.apple.driver.AppleUSBMergeNub       900.4.1
com.apple.driver.AppleUSBHostMergeProperties    1.1
com.apple.driver.usb.AppleUSBHostPacketFilter   1.0
com.apple.iokit.IOSCSIMultimediaCommandsDevice  394
com.apple.iokit.IOBDStorageFamily       1.8
com.apple.iokit.IODVDStorageFamily      1.8
com.apple.iokit.IOCDStorageFamily       1.8
com.apple.iokit.IOAHCISerialATAPI       266
com.apple.iokit.IOSCSIArchitectureModelFamily   394
com.apple.iokit.IO80211Family   1200.12.2
com.apple.driver.mDNSOffloadUserClient  1.0.1b8
com.apple.iokit.IONetworkingFamily      3.2
com.apple.driver.corecapture    1.0.4
com.apple.iokit.IOUSBFamily     900.4.1
com.apple.iokit.IOAHCIFamily    288
com.apple.driver.usb.AppleUSBXHCIPCI    1.1
com.apple.driver.usb.AppleUSBXHCI       1.1
com.apple.driver.usb.AppleUSBEHCIPCI    1.1
com.apple.driver.usb.AppleUSBEHCI       1.1
com.apple.iokit.IOUSBHostFamily 1.1
com.apple.driver.AppleBusPowerController        1.0
com.apple.driver.AppleEFINVRAM  2.1
com.apple.driver.AppleEFIRuntime        2.0
com.apple.iokit.IOHIDFamily     2.0.0
com.apple.security.sandbox      300.0
com.apple.kext.AppleMatch       1.0.0d1
com.apple.driver.AppleKeyStore  2
com.apple.driver.AppleMobileFileIntegrity       1.0.5
com.apple.driver.AppleCredentialManager 1.0
com.apple.driver.DiskImages     444.20.3
com.apple.iokit.IOStorageFamily 2.1
com.apple.iokit.IOReportFamily  31
com.apple.driver.AppleFDEKeyStore       28.30
com.apple.driver.AppleACPIPlatform      5.0
com.apple.iokit.IOPCIFamily     2.9
com.apple.iokit.IOACPIFamily    1.4
com.apple.kec.Libm      1
com.apple.kec.pthread   1
com.apple.kec.corecrypto        1.0

Model: iMac14,2, BootROM IM142.0118.B00, 4 processors, Intel Core i7, 4 GHz, 16 GB, SMC 2.15f2
Graphics: NVIDIA GeForce GT 740, NVIDIA GeForce GT 740, PCIe, 4095 MB
Memory Module: BANK0/DIMM1, 8 GB, DDR3, 1600 MHz, Crucial Technology, BLS8G3D1609DS1S00.
Memory Module: BANK1/DIMM1, 8 GB, DDR3, 1600 MHz, Crucial Technology, BLS8G3D1609DS1S00.
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x111), Broadcom BCM43xx 1.0 (7.21.171.47.1a8)
Bluetooth: Version 5.0.1f7, 3 services, 27 devices, 1 incoming serial ports
Network Service: Ethernet, Ethernet, en0
Network Service: Wi-Fi, AirPort, en1
Serial ATA Device: Samsung SSD 840 EVO 1TB, 1 TB
Serial ATA Device: HL-DT-ST BD-RE  WH10LS30
USB Device: USB 3.0 Bus
USB Device: Drobo
USB Device: USB 2.0 Bus
USB Device: Hub
USB Device: Keyboard Hub
USB Device: Apple Keyboard
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
USB Device: USB 2.0 Bus
USB Device: Hub
USB Device: Hub
USB Device: Hub
USB Device: Flash Card Reader
Thunderbolt Bus: 
```

Yes, it is a hackintosh, so there might be some driver issue somewhere. It was working fine before switching to Sierra (brand new install on the same hardware). Any suggestions/ideas?

Here's a log from transcode_video:
```
[05:59:42] hb_init: starting libhb thread
[05:59:42] thread 70000406e000 started ("libhb")
HandBrake 20161121195327-056b10f-master (2016112301) - Darwin x86_64 - https://handbrake.fr
8 CPUs detected
Opening ../BD/Finding_Dory_t08.mkv...
[05:59:42] CPU: Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz
[05:59:42]  - Intel microarchitecture Haswell
[05:59:42]  - logical processor count: 8
[05:59:42] OpenCL device #1: NVIDIA GeForce GT 740
[05:59:42]  - OpenCL version: 1.2
[05:59:42]  - driver version: 10.12.68 355.10.05.05b16
[05:59:42]  - device type:    GPU
[05:59:42]  - supported:      no
[05:59:42] hb_scan: path=../BD/Finding_Dory_t08.mkv, title_index=1
udfread ERROR: ECMA 167 Volume Recognition failed
disc.c:274: failed opening UDF image ../BD/Finding_Dory_t08.mkv
disc.c:352: error opening file BDMV/index.bdmv
disc.c:352: error opening file BDMV/BACKUP/index.bdmv
[05:59:42] bd: not a bd - trying as a stream/file instead
libdvdnav: Using dvdnav version 5.0.1
libdvdread: Encrypted DVD support unavailable.
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.BUP failed
libdvdread: Can't open file VIDEO_TS.IFO.
libdvdnav: vm: failed to read VIDEO_TS.IFO
...
...
[05:59:43] encx264: min-keyint: 24, keyint: 240
[05:59:43] encx264: encoding at constant RF 1.000000
[05:59:43] encx264: unparsed options: vbv-maxrate=6000:qpmax=34:vbv-bufsize=12000:crf-max=25:level=4.0
x264 [info]: using SAR=1/1
x264 [info]: using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 AVX2 LZCNT BMI2
x264 [info]: profile High, level 4.0
[05:59:43] sync: first pts video is 0
[05:59:43] sync: "Chapter 1" (1) at frame 1 time 0
[05:59:43] sync: first pts audio 0x1 is 0
[05:59:43] sync: first pts audio 0x1 is 0
Encoding: task 1 of 1, 0.03 %
Encoding: task 1 of 1, 0.07 %
...
...
```

**Comment by donmelton** [comment]
@skinless8 Woof. This is a hard one. Since your machine is a hackintosh, could `HandBrakeCLI` be making the wrong assumption about threads and what it's capable of?

You can force the x264 library in `HandBrakeCLI` to use a single thread by passing the `--encoder-option threads=1` on the command line. It will be slow, but maybe it won't kernel panic. And that could tell us something.

**Comment by donmelton** [comment]
@skinless8 Were you able to figure out the problem yet?

**Comment by skinless8** [comment]
@donmelton Thanks for the pointers. With `--encoder-option threads=1`, I was able to complete transcode without kernel panic. (Like you said, it was slow, but it got the job done)

I'll try to update with the latest nightly build of `handbrake` and see if it got resolved. If it does, I'll report back. 

Thank you for your help!

**Comment by hughker** [comment]
@skinless8 good to know about the potential bug with Hackint0sh — my main machine is dying and I was thinking about building one to replace it. But since it primarily does my video transcoding, I might hold off for now...

**Comment by donmelton** [comment]
@skinless8 Good to hear! OK, I'll leave this open for a bit longer and wait for your report. That will also give others a chance to comment.

@hughker Yep, this is always the risk with using non-Apple hardware for macOS. You're smart to hold off.

**Comment by donmelton** [comment]
@skinless8 I'm closing this for now but feel free to continue commenting here.

**Comment by skinless8** [comment]
Solved this many moons ago, but forgot to update here.

As it turned out, doing transcoding is a really good burn-in test for cpu and memory. My "K" cpu was overheating and causing the panic. This was under default optimized setting on the BIOS! I had to make some tweaks on the BIOS to prevent cpu temp to go above 80C.  I also replaced the memory since the memtest found some bad spots.

Now it's transcoding at about 120% of real-time and stable.

Thanks all! This is actually a better real-world test for cpu, memory and tuning configuration, then some of the synthetic tests.
